### PR TITLE
Completed Part 6

### PR DESCRIPTION
**Pull request by derekfairholm**
Completed Part 6

### CODE REVIEW

File: main.cpp [hunk 1]
```diff
@@ -88,6 +89,8 @@ struct Point
     float x{0}, y{0};
 };
 
+using FloatFunctionPtr = void(*)(float&);
+
```
Comment (matkatmusic) [inline main.cpp]: This should be scoped to inside the FloatType class.  it should not be a global alias.
- [x] move alias.

File: main.cpp [hunk 2]
```diff
@@ -137,6 +173,8 @@ FloatType& FloatType::operator/=(float f)
     return *this;
 }
 
+using DoubleFunctionPtr = void(*)(double&);
+
```
Comment (matkatmusic) [inline main.cpp]: This should be scoped to inside the DoubleType class.  it should not be a global alias.
- [x] move alias.

File: main.cpp [hunk 3]
```diff
@@ -189,6 +260,8 @@ DoubleType& DoubleType::operator/=(double d)
     return *this;
 }
 
+using IntFunctionPtr = void(*)(int&);
+
```
Comment (matkatmusic) [inline main.cpp]: This should be scoped to inside the IntType class.  it should not be a global alias.
- [x] move alias.

File: main.cpp [hunk 4]
```diff
@@ -359,7 +464,23 @@ int main()
...
     DoubleType doubleType(3.1);
-    
+
+    // Float Type: Apply() passing a lambda
+
+    floatType.apply([&ft = floatType](float& f) -> FloatType&
+    {
+        f *= 8.f;
+        return ft;
+    });
```
Comment (matkatmusic) [inline main.cpp]: I've never seen anyone capture a reference like this and rename the reference.
Did you do this just to do it, or to improve readability?

**Review by matkatmusic — CHANGES_REQUESTED** [review]

File: main.cpp (hunk 1 above)
Comment (derekfairholm) [reply main.cpp]: I use the same alias in the out of class implementation of `FloatType& FloatType::apply(FloatFunctionPtr);` on line 136. Should I move the function implementation inside the class or not use the alias in the out of class argument list?

**Review by matkatmusic — APPROVED** [review]

**Pull request merged**